### PR TITLE
TestSchemaMerging

### PR DESCRIPTION
**Pull request by rohit-rk**
test_dots_in_property_names: problem forming dbcolumns in setop_table.py.

### CODE REVIEW

**Comment by klahnakoski** [comment]
Update requirements.txt with "mo-dots>=1.4"  The problem is deep in this line:

    acc[literal_field(nested_path)][literal_field(schema.get_column_name(col))][json_type_to_sql_type[col.type]] = quote_column(col.es_column).sql

`acc` is `Data`, and we are using the `literal_field()` call to escape the dots while creating the inner `dict`s that `acc` requires.  The actual error is fixed here:

https://github.com/klahnakoski/mo-dots/commit/e45e5808ac70e7697fd34022af1ed26321f78cb3#diff-33e6d309a1f9a88212b7804f231b0153L99

..and I another test to catch it:  It only happened when the inner `dict` is missing. This line is the same as:

    acc = unwrap(acc)
    tempa = acc.setdefault(nested_path, {})
    tempb = tempa.setdefault(schema.get_column_name(col), {})
    tempb[json_type_to_sql_type[col.type]] = quote_column(col.es_column).sql

**Comment by rohit-rk** [comment]
Oh yes, I see. thanks. Updated the branch, this part works now.

File: jx_sqlite/setop_table.py [hunk 1]
```diff
@@ -347,17 +347,15 @@ def _accumulate_nested(rows, row, nested_doc_details, parent_doc_id, parent_id_c
...
             data = result.data
-
+        
         if query.format == "cube":
             for f, _ in self.sf.tables.items():
-                if frum.endswith(f):  
-                    data = result.data
-                    
-                    num_rows = len(data)
+                if frum.endswith(f) or (test_dots(cols) and isinstance(query.select, list)):
```
Comment (rohit-rk) [inline jx_sqlite/setop_table.py]: I hope this is okay.

File: pyLibrary/queries/expressions.py [hunk 2]
```diff
@@ -288,18 +288,27 @@ def to_sql(self, schema, not_null=False, boolean=False):
                     value = quote_column(col.es_column).sql
                 else:
                     value = "(" + quote_column(col.es_column).sql + ") IS NOT NULL"
-                acc[literal_field(nested_path)][literal_field(schema.get_column_name(col))]['b'] = value
+                acc = unwrap(acc)
```
Comment (klahnakoski) [inline pyLibrary/queries/expressions.py]: don't actually do this; initialize `acc = {}`

File: jx_sqlite/setop_table.py (hunk 1 above)
Comment (klahnakoski) [inline jx_sqlite/setop_table.py]: this whole loop is dubious, but wee will merge and refactor later

**Pull request merged**